crates/shielded_token/src/masp/shielded_wallet.rs

```diff
@@ -2076,13 +2035,12 @@ mod test_shielded_wallet {
         // to construct conversions properly, just like the protocol does, i.e.
         // with conversions for non-consecutive epochs
 
-        // Query the balance with conversions at epoch 4
+        // Query the balance with conversions at epoch 5
```

Comment:
Probably this test is not needed anymore since it's testing the functionality that this pr removes?

Reply:
Thanks, you're right. But maybe this test can just be repurposed since the specific checks that it does are still useful in the absence of inverse conversions...

Reply:
Right but I think for this test to make sense we should shield at epoch 4 instead of 5. Right now we shield at 5 and query at 5 so we just get back the same amount. We should instead shield at 4, query at epoch 4 expecting the same amount and then query at epoch 5 expecting the rewards

Reply:
Understood, thanks. The shielding of 10 NAM now happens at epoch 1, and computing the exchanged amount yields 160 NAM of epoch 5. This seems to line up with the conversion -2NAM@ep1 + 32NAM@ep5 being applied 5 times. That being said, `compute_exchanged_amount` should not be affected by the value of the current MASP epoch (since I don't think it queries this value). It should really only be affected by what's currently in the allowed conversion Merkle tree that's constructed on lines 1999 to 2014, I think.